# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/remoteworkers/RemoteWorkerChild.cpp
# Commit: fb720435215f
# Full Hash: fb720435215fa5be3b2bd2101f3a54382954b70c
# Author: Andrew Sutherland <asutherland@asutherland.org>
# Date: 2024-10-24 09:44:34
# Regressor Bug: 1544232
# File Overlap Count: 4
# Description:
#   Bug 1544232 - Limit lifetime extension of SWs by SWs to the sender's lifetime. r=edenchuang
#   
#   This patch introduces an explicit concept of lifetimes with mechanisms
#   in place so that actions taken by Clients (windows or non-ServiceWorker
#   orkers) will extend the lifetime of a ServiceWorker, but a ServiceWorker
# ==============================================================================

diff -r 25452f6623a0 -r fb720435215f dom/workers/remoteworkers/RemoteWorkerChild.cpp
--- a/dom/workers/remoteworkers/RemoteWorkerChild.cpp	Thu Oct 24 03:02:41 2024 +0000
+++ b/dom/workers/remoteworkers/RemoteWorkerChild.cpp	Thu Oct 24 03:02:42 2024 +0000
@@ -323,14 +323,18 @@
 
   nsresult rv = NS_OK;
 
-  if (clientInfo.isSome()) {
-    Maybe<mozilla::ipc::CSPInfo> cspInfo = clientInfo.ref().GetCspInfo();
-    if (cspInfo.isSome()) {
-      info.mCSP = CSPInfoToCSP(cspInfo.ref(), nullptr);
-      info.mCSPInfo = MakeUnique<CSPInfo>();
-      rv = CSPToCSPInfo(info.mCSP, info.mCSPInfo.get());
-      if (NS_WARN_IF(NS_FAILED(rv))) {
-        return rv;
+  if (mIsServiceWorker) {
+    info.mSourceInfo = clientInfo;
+  } else {
+    if (clientInfo.isSome()) {
+      Maybe<mozilla::ipc::CSPInfo> cspInfo = clientInfo.ref().GetCspInfo();
+      if (cspInfo.isSome()) {
+        info.mCSP = CSPInfoToCSP(cspInfo.ref(), nullptr);
+        info.mCSPInfo = MakeUnique<CSPInfo>();
+        rv = CSPToCSPInfo(info.mCSP, info.mCSPInfo.get());
+        if (NS_WARN_IF(NS_FAILED(rv))) {
+          return rv;
+        }
       }
     }
   }